### PR TITLE
US969005: Initial commit

### PR DESCRIPTION
**Pull request by rorytorneymf**
https://internal.almoctane.com/ui/entity-navigation?p=131002/6001&entityType=work_item&id=969005

### CODE REVIEW

**Comment by rorytorneymf** [comment]
You might need to adjust the perms for this new repo like for caf-ssl:

https://internal.almoctane.com/ui/entity-navigation?p=131002/6001&entityType=work_item&id=887107

image

**Comment by rorytorneymf** [comment]
I wasn't sure if this should be a module in [caf-common](https://github.com/CAFapi/caf-common) rather than a new repo? I created a new repo as that what I did previously for [caf-ssl](https://github.com/CAFapi/caf-ssl). 

(If a new repo is the right thing, then it might make sense to deprecate the [caf-ssl](https://github.com/CAFapi/caf-ssl) repo and move the [caf-ssl-dropwizard](https://github.com/CAFapi/caf-ssl/tree/main/caf-ssl-dropwizard) module into this repo?)

**Comment by dermot-hardy** [comment]
> You might need to adjust the perms...

Done

> I wasn't sure if this should be a module in [caf-common](https://github.com/CAFapi/caf-common) rather than a new repo? I created a new repo...

I don't think it matters too much but I probably would have went for a new repo too.  Once this is released I don't see it changing too often.

> ...it might make sense to deprecate the [caf-ssl](https://github.com/CAFapi/caf-ssl) repo and move the [caf-ssl-dropwizard](https://github.com/CAFapi/caf-ssl/tree/main/caf-ssl-dropwizard) module into this repo?

Ok we can raise that as a separate ticket but it would be low-priority as there isn't really an impetus for us to do it.

File: official-build.props [hunk 1]
```diff
@@ -0,0 +1 @@
+SEPG_BUILD_ENV_IMAGE=cafapi/buildenv-jdk17:1.5.1
```
Comment (dermot-hardy) [inline official-build.props]: Update to the latest:

```suggestion
SEPG_BUILD_ENV_IMAGE=cafapi/buildenv-jdk17:4.1.0
```

File: caf-substitutor-dropwizard/src/main/java/com/github/cafapi/util/dropwizard/CustomSubstitutor.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import com.hpe.caf.secret.SecretUtil;
+
+/**
+ * A custom implementation of StringSubstitutor that provides additional functionality
+ * for variable substitution, including secret lookup.
+ * <p>
+ * This class extends StringSubstitutor and configures it with a custom StringLookup
+ * that combines secret lookup and environment variable lookup capabilities.
+ */
+public final class CustomSubstitutor extends StringSubstitutor
```
Comment (dermot-hardy) [inline caf-substitutor-dropwizard/src/main/java/com/github/cafapi/util/dropwizard/CustomSubstitutor.java]: Before I looked at this I thought it should be called something like `EnvironmentVariableAndSecretSubstitutor` but now that I've looked at how generic and extendable the implementation that you've written is I'm wondering if something like `CafConfigSubstitutor` would be more appropriate?

**Review by dermot-hardy — APPROVED** [review]

**Comment by rorytorneymf** [comment]
> > You might need to adjust the perms...
> 
> Done
> 
> > I wasn't sure if this should be a module in [caf-common](https://github.com/CAFapi/caf-common) rather than a new repo? I created a new repo...
> 
> I don't think it matters too much but I probably would have went for a new repo too. Once this is released I don't see it changing too often.
> 
> > ...it might make sense to deprecate the [caf-ssl](https://github.com/CAFapi/caf-ssl) repo and move the [caf-ssl-dropwizard](https://github.com/CAFapi/caf-ssl/tree/main/caf-ssl-dropwizard) module into this repo?
> 
> Ok we can raise that as a separate ticket but it would be low-priority as there isn't really an impetus for us to do it.

I have raised [US975352 Consolidate Dropwizard utility repos](https://internal.almoctane.com/ui/entity-navigation?p=131002/6001&entityType=work_item&id=975352)

**Pull request merged**